assets/js/controllers/saved_stops_index_controller.js

```diff
@@ -0,0 +1,54 @@
+import { Controller } from "stimulus"
+import Storage from "../storage"
+
+// The controller fetches all saved stops and displays the link
+// to the "/live" page and a button to remove it.
+// If there are no saved stops, they get a message telling them that.
+export default class extends Controller {
+  connect() {
+    const stops = Storage.getSavedStops();
+
+    if(Object.keys(stops).length > 0) {
+      this.renderStops(stops);
+    } else {
+      this.renderNoStops();
+    }
+  }
+
+  // This makes use of the HTML <template> tag, and appends a child to the controller div
+  // for each of the saved stops. The button gets a data-key attribute set with the value
+  // of the key from the stored stops "database". #remove uses this id in the click callback
+  // to delete it from the "database".
+  renderStops(stops) {
+    const template = document.querySelector('#saved-stops');
+    for (let [key, stop] of Object.entries(stops)) {
+      const clone = template.content.cloneNode(true);
+      const anchor = clone.querySelector("a");
+      const strong = clone.querySelector("strong");
+      const button = clone.querySelector("button");
+
+      strong.textContent = stop.name;
+      anchor.href = `/live?stop_ids=${key}`;
+      button.textContent = "Remove";
+      button.setAttribute(`data-key`, `${key}`)
+      button.setAttribute('data-action', `click->${this.identifier}#remove`)
+      this.element.appendChild(clone);
```

Comment:
This part feels a little funky, since it relies on accessing unique tags within the template. You could create another Stimulus child controller that is the root element in the template, pass in the key and name as data attributes, and then make the new controller responsible for setting up the name, data-key, and href using data targets. The click action can still refer to the parent controller and can be baked into the template since it will be the same for all children.

Reply:
I do like this for a future change, but I think I want to err on the side of keeping things in one place for now.